src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,102 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_CAPACITY = 16;
+    private static final double LOAD_FACTOR = 0.75;
+    private static final int SIZE_MULTIPLIER = 2;
+    private int size;
+    private Node<K,V>[] table;
+
+    public MyHashMap() {
+        table = (Node<K, V>[]) new Node[DEFAULT_CAPACITY];
+    }
 
     @Override
     public void put(K key, V value) {
-
+        if (size >= (int) (table.length * LOAD_FACTOR)) {
+            resize();
+        }
+        Node<K, V> indexNode = table[getIndex(key)];
+        Node<K, V> newNode = new Node<>(key, value);
+        if (indexNode == null) {
+            table[getIndex(key)] = newNode;
+        } else {
+            getLastNode(key, value, indexNode, newNode);
+        }
+        size++;
     }
 
     @Override
     public V getValue(K key) {
-        return null;
+        Node<K, V> node = getNode(key);
+        return node == null ? null : node.value;
     }
 
     @Override
     public int getSize() {
-        return 0;
+        return size;
+    }
+
+    private Node<K, V> getNode(K key) {
+        Node<K, V> node = table[getIndex(key)];
+        if (node == null) {
+            return null;
+        }
+        while (!Objects.equals(node.key, key)) {
+            node = node.next;
+        }
+        return node;
+    }
+
+    private void getLastNode(K key, V value, Node<K, V> indexNode, Node<K, V> newNode) {
```

Comment:
`getLastNode` means that you should return last node. But you just put new node as a last node, so rename this method to `putAtTheEnd` or sth like this.